src/pugixml.cpp

```diff
@@ -3160,6 +3177,8 @@ PUGI__NS_BEGIN
 
 			if (declaration ? PUGI__OPTSET(parse_declaration) : PUGI__OPTSET(parse_pi))
 			{
+				char_t ch = 0;
```

Comment:
I am using cppcheck for all of my projects. This declaration generates a warning today because it's scope can be reduced. My suggestion would be the keep this change in order to minimize the number of warnings. It would be very easy to move the declaration in the future if it makes any sense.

But anyway, I have reverted the change in order to comply with your rules.